kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -262,6 +264,27 @@ impl<'tcx> GotocCtx<'tcx> {
         self.reset_current_fn();
     }
 
+    pub fn is_proof_harness(&self, def_id: DefId) -> bool {
+        let all_attributes = self.tcx.get_attrs_unchecked(def_id);
```

Comment:
What does the `unchecked` imply here? There seems to be a comment that this function will be removed: https://github.com/rust-lang/rust/blob/6153d3cbe6abc74fb37e4ebe48cc825484fd6bbf/compiler/rustc_middle/src/ty/mod.rs#L2352

Reply:
Yeah, we already use this function to process Kani attributes. I remember bumping into this name change in one of our rustc upgrades (https://github.com/model-checking/kani/pull/1209). Looking at the original rustc change (https://github.com/rust-lang/rust/pull/95562), someone did raise the point that removing this function would be a problem for tool attributes. The conclusion seems to be that in order to remove this function, `rustc` should provide an alternative to get tool attributes. See thread here:
https://github.com/rust-lang/rust/pull/95562/files#r914276675